apps/prairielearn/src/question-servers/calculation-subprocess.js

```diff
@@ -36,20 +38,29 @@ async function prepareChunksIfNeeded(question, course) {
 
 async function callFunction(func, question_course, question, inputData) {
   await prepareChunksIfNeeded(question, question_course);
-  const coursePath = chunks.getRuntimeDirectoryForCourse(question_course);
+
+  const courseHostPath = chunks.getRuntimeDirectoryForCourse(question_course);
+  const courseRuntimePath = config.workersExecutionMode === 'native' ? courseHostPath : '/course';
+
   const { fullPath: questionServerPath } = await filePaths.questionFilePathAsync(
     'server.js',
     question.directory,
-    coursePath,
+    courseHostPath,
     question,
   );
+
+  // The worker will use the non-host path, which may be different.
+  // Rework the paths to be relative to the worker's filesystem.
+  const questionServerPathWithinCourse = path.relative(courseHostPath, questionServerPath);
+  const questionServerRuntimePath = path.join(courseRuntimePath, questionServerPathWithinCourse);
+
   try {
-    return withCodeCaller(coursePath, async (codeCaller) => {
-      const res = await codeCaller.call('v2-question', null, questionServerPath, null, [
+    return await withCodeCaller(courseHostPath, async (codeCaller) => {
```

Comment:
Missing `await`, which means the enclosing `try` was never triggered. This wasn't *actually* a huge problem in practice, as the caller would still handler the error and pass it back upstream.

Reply:
I know this isn't a `.ts` file, but could we turn on some or all of:
* https://typescript-eslint.io/rules/no-floating-promises
* https://typescript-eslint.io/rules/promise-function-async
* https://typescript-eslint.io/rules/return-await

Reply:
See #6801.

Reply:
Oh right, I knew about this but totally forgot. Thanks!